queue/src/main/java/org/killbill/queue/dispatching/Dispatcher.java

```diff
@@ -94,25 +100,35 @@ public CallableQueue(final M entry, final CallableCallback<E, M> callback) {
 
         @Override
         public E call() throws Exception {
-            final E event = callback.deserialize(entry);
-            if (event != null) {
-                Throwable lastException = null;
-                long errorCount = entry.getErrorCount();
-                try {
-                    callback.dispatch(event, entry);
-                } catch (final Exception e) {
-                    if (e.getCause() != null && e.getCause() instanceof InvocationTargetException) {
-                        lastException = e.getCause().getCause();
-                    } else {
+            try {
+                final UUID userToken = entry.getUserToken();
+                MDC.put(MDC_KB_USER_TOKEN, userToken != null ? userToken.toString() : null);
```

Comment:
So, will this be logged automatically or does this require changes in the log format as well?

Reply:
To close the loop on this: https://github.com/killbill/killbill/pull/896#discussion_r174279072.